### PR TITLE
TPC CalDet wrapper

### PR DESCRIPTION
**Pull request by tklemenz**
- Added a wrapper for the TPC CalDet class so that CalDet objects can be published to the CCDB. This is needed only temporarily and will be removed at a later point.

- Added a check in the /= operator of the CalArray class to prevent division by 0

### CODE REVIEW

File: Detectors/TPC/base/include/TPCBase/CalArray.h [hunk 1]
```diff
@@ -247,7 +247,9 @@ inline const CalArray<T>& CalArray<T>::operator/=(const CalArray<T>& other)
     return *this;
   }
   for (size_t i = 0; i < mData.size(); ++i) {
-    mData[i] /= other.getValue(i);
+    if (other.getValue(i) != 0) {
+      mData[i] /= other.getValue(i);
+    }
```
Comment (wiechula) [inline Detectors/TPC/base/include/TPCBase/CalArray.h:252]: Perhaps you should print an error stating that division by 0 was detected and set the value to 0

File: Detectors/TPC/qc/include/TPCQC/CalPadWrapper.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  {
+    return mObj;
+  }
+
+  virtual const char* GetName() const override { return mObj ? mObj->getName().data() : "unset"; }
+
+  virtual ~CalPadWrapper() override = default;
+
+ private:
+  o2::tpc::CalDet<float>* mObj{};
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/CalPadWrapper.h:52]: Please add a comment

File: Detectors/TPC/qc/include/TPCQC/CalPadWrapper.h [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+namespace o2
+{
+namespace tpc
+{
+namespace qc
+{
+/// A wrapper class to easily promote any type to a TObject
+/// does not take ownership of wrapped object and should not be used
+/// in tight loops since construction expensive
```
Comment (wiechula) [inline Detectors/TPC/qc/include/TPCQC/CalPadWrapper.h:27]: Please add a statement that this is a temporary solution until QualityControl can handle objects which don't inherit from TObject.

**Comment by tklemenz** [comment]
Hi @wiechula, I implemented your suggestions.

**Comment by stheckel** [comment]
Dear @davidrohr and @shahor02 
could you have a look please? I think, this PR can be merged and two other commits (in QC) are waiting for it.
Thanks and cheers,
Stefan

**Pull request merged**